File: components/ExternalLink.js
import React from "react";
import css from "styled-jsx/css";

const ExternalLink = ({ children, to = "" }) => (
	<div className="external-link-container">
		<a target="_blank" rel="noopener noreferrer" href={to} className="external-link">
			{children}
		</a>
		<style jsx>{styles}</style>
	</div>
);

const styles = css`
	.external-link-container {
		display: inline-block;
	}

	.external-link {
		display: block;
		border: none;
		background: none;
		cursor: pointer;
		box-sizing: border-box;
		padding: 4px 10px;
		border-radius: 10px;
		text-decoration: none;
	}

	.external-link:hover {
		opacity: 0.7;
		background: #f0f8ff;
	}
`;

export default ExternalLink;

File: src/components/external_link.rs
use yew::prelude::*;

use super::text::Text;

pub struct ExternalLink {
	props: Props,
}

#[derive(Clone, PartialEq, Properties)]
pub struct Props {
	pub to: String,
	pub children: ChildrenWithProps<Text>,
}

impl Component for ExternalLink {
	type Message = ();
	type Properties = Props;

	fn create(props: Self::Properties, _: ComponentLink<Self>) -> Self {
		Self { props }
	}

	fn update(&mut self, _: Self::Message) -> ShouldRender {
		false
	}

	fn change(&mut self, _: Self::Properties) -> ShouldRender {
		false
	}

	fn view(&self) -> Html {
		html! {
			<div class="external-link-container">
				<a target="_blank" rel="noopener noreferrer" href=self.props.to.to_string() class="external-link">
					{ self.props.children.clone()}
				</a>
			</div>
		}
	}
}
